packages/atomic/src/components/facets/atomic-segmented-facet-scrollable/atomic-segmented-facet-scrollable.tsx

```diff
@@ -21,32 +33,107 @@ type ArrowDirection = 'right' | 'left';
   styleUrl: 'atomic-segmented-facet-scrollable.pcss',
   shadow: true,
 })
-export class AtomicSegmentedFacetScrollable {
+export class AtomicSegmentedFacetScrollable implements InitializableComponent {
+  @InitializeBindings()
+  public bindings!: Bindings;
+  public searchStatus!: SearchStatus;
+  @BindStateToController('searchStatus')
+  @State()
+  public searchStatusState!: SearchStatusState;
+  @State()
+  public error!: Error;
+
   private horizontalScroll?: HTMLDivElement;
+  @State() private hideLeftArrow = true;
+  @State() private hideRightArrow = false;
+  private observer!: ResizeObserver;
+
+  public initialize() {
+    this.searchStatus = buildSearchStatus(this.bindings.engine);
+  }
+
+  componentDidLoad() {
+    this.observer = new ResizeObserver(() => {
+      this.handleScroll();
+    });
+    this.observer.observe(this.horizontalScroll as HTMLDivElement);
+  }
+
+  disconnectedCallback() {
+    this.observer.disconnect();
+  }
+
+  @Listen('mousewheel')
```

Comment:
Why listen to the mouse wheel event? I don't think this will work on mobile
You can try it out with the dev tools
<img width="249" alt="Screen Shot 2022-07-04 at 11 51 56 AM" src="https://user-images.githubusercontent.com/4923043/177187773-db2409d6-74ec-4c10-a87e-2fa2d31f3b28.png">

Reply:
Perhaps the `touchmove` event.

Reply:
ok, we actually need both !